internal/controller/eventing/nats_test.go

```diff
@@ -0,0 +1,216 @@
+package eventing
+
+import (
+	"errors"
+	"testing"
+
+	"github.com/kyma-project/eventing-manager/pkg/env"
+	managermocks "github.com/kyma-project/eventing-manager/pkg/eventing/mocks"
+	subscriptionmanagermocks "github.com/kyma-project/eventing-manager/pkg/subscriptionmanager/mocks"
+	ecsubmanagermocks "github.com/kyma-project/eventing-manager/pkg/subscriptionmanager/mocks/ec"
+	"github.com/kyma-project/eventing-manager/test/utils"
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+)
+
+func Test_reconcileNATSSubManager(t *testing.T) {
+	t.Parallel()
+
+	// given - common for all test cases.
+	givenEventing := utils.NewEventingCR(
+		utils.WithEventingStreamData("Memory", "650M", 99, 98),
+		utils.WithEventingEventTypePrefix("one.two.three"),
+	)
+
+	givenNATSConfig := env.NATSConfig{
+		URL:                     "test123",
+		MaxReconnects:           10,
+		ReconnectWait:           11,
+		EventTypePrefix:         "12",
+		MaxIdleConns:            13,
+		MaxConnsPerHost:         14,
+		MaxIdleConnsPerHost:     15,
+		IdleConnTimeout:         16,
+		JSStreamName:            "17",
+		JSSubjectPrefix:         "18",
+		JSStreamStorageType:     "19",
+		JSStreamReplicas:        20,
+		JSStreamRetentionPolicy: "21",
+		JSStreamMaxMessages:     22,
+		JSStreamMaxBytes:        "23",
+		JSStreamMaxMsgsPerTopic: 24,
+		JSStreamDiscardPolicy:   "25",
+		JSConsumerDeliverPolicy: "26",
+	}
+
+	givenBackendConfig := &env.BackendConfig{
+		DefaultSubscriptionConfig: env.DefaultSubscriptionConfig{
+			MaxInFlightMessages:   55,
+			DispatcherRetryPeriod: 56,
+			DispatcherMaxRetries:  57,
+		},
+	}
+
+	// define test cases
+	testCases := []struct {
+		name                         string
+		givenIsNATSSubManagerStarted bool
+		givenNATSSubManagerMock      func() *ecsubmanagermocks.Manager
+		givenEventingManagerMock     func() *managermocks.Manager
+		givenManagerFactoryMock      func(*ecsubmanagermocks.Manager) *subscriptionmanagermocks.ManagerFactory
+		wantAssertCheck              bool
+	}{
+		{
+			name:                         "should do nothing if NATSSubManager is already started",
```

Comment:
Nit: Would it be possible to reformulate the test cases somewhat along the format `expectation because of setup`? 

For example:
nothing happens/reconciler does nothing because NATSSubManager is already started

Reply:
I changed it to refer as `it`, because its a unit test and its referring to method `reconcileNATSSubManager`, not the whole reconciler.